solutions/cf-609E.mdx

```diff
@@ -0,0 +1,179 @@
+---
+id: cf-609E
+source: CF
+title: Minimum spanning tree for each edge
+author: Chongtian Ma
+---
+
+[Official Analysis](https://codeforces.com/blog/entry/22187)
+
+## Implementation 
+
+As mentioned in the official editorial, the answer for each edge is removing the heaviest edge in the MST in
+the path from vertex $u$ to $v$ and add the given weight. This will ensure that the MST is still connected.
+
+**Time Complexity:** $\mathcal{O}(M\log N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+
+const int MAXN = 2e5 + 1;
+const int MAXL = 20; // approximately maximum log
+
+// for euler-tour
+array<int, MAXN> enter_time, exit_time, depth;
+
+// for binary lifting
+array<array<int, MAXL>, MAXN> up, max_up;
+// up[i][j] stores the 2^jth ancestor of vertex i
+// max_up[i][j] stores the maximum weight of all the edges
+// from i to its 2^jth ancestor
```

Comment:
1. put these above the variable declarations
2. separate lines for var declarations